pandas/core/nanops.py

```diff
@@ -1332,31 +1332,24 @@ def nancorr(
 
 
 def get_corr_func(method):
-    if method in ["kendall", "spearman"]:
-        from scipy.stats import kendalltau, spearmanr
-    elif method in ["pearson"]:
-        pass
+    if method == "kendall":
+        from scipy.stats import kendalltau
+
+        return lambda a, b: kendalltau(a, b)[0]
+    elif method == "spearman":
+        from scipy.stats import spearmanr
+
+        return lambda a, b: spearmanr(a, b)[0]
+    elif method == "pearson":
+        return lambda a, b: np.corrcoef(a, b)[0, 1]
```

Comment:
can we do this without the lambda?  makes for nicer tracebacks/profiling etc

Reply:
Updated